### PR TITLE
[SharedWithYou] Added support for Xcode 14.1 b3

### PR DESCRIPTION
**Pull request by SotoiGhost**
* Added macOS support
* Added some missing attributes and a class

### CODE REVIEW

File: src/sharedwithyou.cs [hunk 1]
```diff
@@ -188,23 +188,41 @@ interface SWHighlightCenter
...
+
+		[NoTV]
 		[Async]
 		[Export ("getSignedIdentityProofForCollaborationHighlight:usingData:completionHandler:")]
 		void GetSignedIdentityProof (SWCollaborationHighlight collaborationHighlight, NSData data, Action<SWSignedPersonIdentityProof, NSError> completionHandler);
 	}
 
-
+#if __MACOS__
+	[NoWatch, Mac (13,0), NoTV, NoiOS, NoMacCatalyst]
```
Comment (stephen-hawley) [inline src/sharedwithyou.cs:216]: Q: If this is in a `__MACOS__` test, do you need `NoWatch`, `NoTV`, `NoiOS` and `NoMacCatalyst`?

Comment (rolfbjarne) [reply src/sharedwithyou.cs:216]: Ideally we wouldn't have any #if ... inside api definitions, so if it could be removed, that would be much better.

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Looks like this needs SharedWithYouCore first?

> sharedwithyou.cs(16,7): error CS0246: The type or namespace name 'SharedWithYouCore' could not be found (are you missing a using directive or an assembly reference?)

**Comment by mandel-macaque** [comment]
> Looks like this needs SharedWithYouCore first?
> 
> > sharedwithyou.cs(16,7): error CS0246: The type or namespace name 'SharedWithYouCore' could not be found (are you missing a using directive or an assembly reference?)

I'm taking care of it

**Pull request merged**